dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/file/ImmediateFileWriter.kt

```diff
@@ -72,13 +74,33 @@ internal class ImmediateFileWriter<T : Any>(
         }
     }
 
+    @Suppress("NestedBlockDepth")
     private fun writeDataToFile(file: File, dataAsByteArray: ByteArray) {
-        val fileSize = file.length()
-        FileOutputStream(file, true).use {
-            if (fileSize > 0) {
-                it.write(separatorBytes)
+        try {
+            val outputStream = FileOutputStream(file, true)
+            outputStream.use { stream ->
+                tryToLockAndWriteBytes(stream, dataAsByteArray)
+            }
+        } catch (e: IllegalStateException) {
+            sdkLogger.e(
+                "Exception when trying to lock the file: [${file.canonicalPath}] ", e
+            )
+        } catch (e: IOException) {
+            sdkLogger.e(
+                "Exception when trying to write data to: [${file.canonicalPath}] ", e
+            )
+        }
+    }
+
+    private fun tryToLockAndWriteBytes(
```

Comment:
why `tryTo`, since there's no try block in here ? I'd rename that `lockFileAndWriteData` (for consistency)